File: manila/src/main/java/cz/fb/manaus/manila/BestChanceLayBettor.java
package cz.fb.manaus.manila;

import cz.fb.manaus.core.model.Side;
import cz.fb.manaus.reactor.betting.listener.AbstractUpdatingBettor;
import cz.fb.manaus.reactor.betting.validator.Validator;
import cz.fb.manaus.spring.ManausProfiles;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
@Profile(ManausProfiles.DB_PROFILE)
public class BestChanceLayBettor extends AbstractUpdatingBettor {

    @ManilaBet
    @Autowired
    public BestChanceLayBettor(List<Validator> validators,
                               BestChanceLayAdviser adviser) {
        super(Side.LAY, validators, adviser);
    }
}

File: manila/src/main/java/cz/fb/manaus/manila/BestChanceLayBettor.kt
package cz.fb.manaus.manila

import cz.fb.manaus.core.model.Side
import cz.fb.manaus.reactor.betting.listener.AbstractUpdatingBettor
import cz.fb.manaus.reactor.betting.validator.Validator
import cz.fb.manaus.spring.ManausProfiles
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

@Component
@Profile(ManausProfiles.DB)
class BestChanceLayBettor @ManilaBet @Autowired
constructor(validators: List<Validator>, adviser: BestChanceLayAdviser) :
        AbstractUpdatingBettor(
                side = Side.LAY,
                validators = validators,
                priceAdviser = adviser)
